src/Appwrite/Platform/Tasks/ScheduleFunctions.php

```diff
@@ -40,32 +40,39 @@ protected function enqueueResources(Group $pools, Database $dbForConsole): void
 
         $delayedExecutions = []; // Group executions with same delay to share one coroutine
 
-        foreach ($this->schedules as $key => $schedule) {
-            $cron = new CronExpression($schedule['schedule']);
-            $nextDate = $cron->getNextRunDate();
-            $next = DateTime::format($nextDate);
+        foreach ($this->schedules as $scheduleKey => $schedule) {
+            if (CronExpression::isValidExpression($schedule['schedule'])) {
+                $cron = new CronExpression($schedule['schedule']);
+                $nextDate = $cron->getNextRunDate();
+            } else {
+                try {
+                    $nextDate = new \DateTime($schedule['schedule']);
+                    $schedule['delete'] = true;
+                } catch (\Exception) {
+                    Console::error('Failed to parse schedule: ' . $schedule['schedule']);
+                    continue;
+                }
+            }
 
+            $next = DateTime::format($nextDate);
             $currentTick = $next < $timeFrame;
 
             if (!$currentTick) {
                 continue;
             }
 
-            $total++;
-
-            $promiseStart = \time(); // in seconds
-            $executionStart = $nextDate->getTimestamp(); // in seconds
-            $delay = $executionStart - $promiseStart; // Time to wait from now until execution needs to be queued
+            $total += 1;
+            $delay = $nextDate->getTimestamp() - \time(); // Time to wait from now until execution needs to be queued
 
             if (!isset($delayedExecutions[$delay])) {
                 $delayedExecutions[$delay] = [];
             }
 
-            $delayedExecutions[$delay][] = $key;
+            $delayedExecutions[$delay][] = $scheduleKey;
```

Comment:
Where is $scheduleKey defined? I cant seem to find it

Reply:
In the foreach above